### PR TITLE
Bug fixes for reflection probes normalization

### PR DESCRIPTION
**Pull request by tlaedre**
### Purpose of this PR

Fixes NaNs in lighting when probe positions are invalid or when probes extraction fails.
Fixed missing probe requests after domain reload.

How to repro NaNs:
Make sure probe volumes and reflection probe normalization is enabled in the project. Place a reflection probe inside geometry. Bake it. Lighting is now broken with nans propagating out from the reflection probe.

How to repro missing requests:
Add a new reflection probe. Do not bake. Force a domain reload (code compilation, e.g.). Bake. Observe how the new probe does not have valid reflection normalization available / applied.

(Need to run full probe volume bakes for this to work, not just baking single reflection probes)

### Testing status

Tested in editor HDRP with PV and reflection normalization enabled.
Only affects baked probes.

### CODE REVIEW

File: com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeReferenceVolume.cs [hunk 1]
```diff
@@ -121,11 +128,10 @@ private bool ComputeRequestIsFree(Vector3 capturePosition)
         /// <returns>Whether the request for light probe rendering has been fulfilled and sh is valid.</returns>
         public bool RetrieveProbeSH(int requestID, out SphericalHarmonicsL2 sh)
         {
-            if (requestID >= 0 && requestID < m_SHCoefficients.Count
-                && ComputeCapturePositionIsValid(m_RequestPositions[requestID]))
+            if (requestID >= 0 && requestID < m_SHCoefficients.Count && ComputeCapturePositionIsValid(m_RequestPositions[requestID]))
             {
                 sh = m_SHCoefficients[requestID];
-                return true;
+                return m_SHValidity[requestID] == kValidSH;
```
Comment (tlaedre) [inline com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeReferenceVolume.cs:134]: Not 100% certain if we should require full validity or also allow a threshold of error here.

**Comment by sebastienlagarde** [comment]
FYI there is also this PR which do a fix in probe normalization, bringing here for awareness: https://github.com/Unity-Technologies/Graphics/pull/6140

File: com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeReferenceVolume.cs (hunk 1 above)
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeReferenceVolume.cs:134]: It really depends on the values returned by the lightmapper on non-completely invalid probe locations; I'd say for now we can force it to be required to be fully valid.

**Comment by FrancescoC-unity** [comment]
> FYI there is also this PR which do a fix in probe normalization, bringing here for awareness: #6140

This is a different issue, it is for when reflection probes are inside or partially inside geometry.

**Review by FrancescoC-unity — APPROVED** [review]

**Review by JulienIgnace-Unity — APPROVED** [review]

**Review by iM0ve — CHANGES_REQUESTED** [review]
**Test status:** Done
**What's tested:**
* Verified the fixes - reproduced the issue and checked that its no longer occurring with this PR
* Regression test for APV baking process - opened the ClassRoom scene and rebaked lighting to see that APV and reflection probes bake as expected
* Regression test for non-APV features - Opened HDRP template and Amalienborg to check if non-APV probes had been affected.
* Functional testing - Added new reflection probes, baked individually and via lighting panel. Also checked Realtime on enable ones.

**Issues:**
1) **FIXED** I can still reproduce the 1st one with nans. 
https://user-images.githubusercontent.com/36502659/141264276-91712911-8a54-4fc1-abde-9190220e2cab.mp4

**Review by iM0ve — APPROVED** [review]
Updated my review

**Pull request merged**